chore(flags): Schedule update of feature flag last_called_at

haacked opened the pull request:
This is a follow-up to https://github.com/PostHog/posthog/pull/38962

This creates a Celery schedule to run a task to update the `last_called_at` column for every feature flag based on `$feature_flag_called` events in Clickhouse.

The task runs every 30 minutes.

## Problem

We want to start tracking the last time a flag was called.

## Changes

Adds a Celery task that runs every 30 minutes. It uses redis locking to ensure only one instance is running at a time. It runs a clickhouse query to get the last time a flag was called looking at `$feature_flag_called` events and updates the new `last_called_at` column with this information. This is not yet exposed to users.

## How did you test this code?

Manually.

<!-- Docs reminder: If this change requires updated docs, please do that! Engineers are the primary people responsible for their documentation. 🙌 -->

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Changelog: (features only) Is this feature complete?

<!-- Optional, but helpful for our content team! -->
<!-- Yes if this is okay to go in the changelog. No if it's still hidden behind a feature flag, or part of a feature that's not complete yet, etc.  -->

haacked committed:
Schedule update of feature flag last_called_at

This creates a Celery schedule to run a task to update the `last_called_at` column for every feature flag based on `$feature_flag_called` events in Clickhouse.

The task runs every 30 minutes.

haacked committed:
Ensure parsed datetime is timezone-aware

Use timezone.make_aware() for naive datetimes parsed from Redis to prevent
TypeError when comparing with timezone.now(). This handles edge cases where
naive datetime strings might be stored during testing or due to bugs.

dustinbyrne commented on posthog/tasks/tasks.py:
```diff
@@ -930,6 +930,187 @@ def background_delete_model_task(
+                )
+                updated_count = len(flags_to_update)
+            except Exception as e:
+                capture_exception(e, additional_properties={"flags_count": len(flags_to_update)})
```
I'd recommend tagging the exception in case you ever need to look it up in PostHog. E.g.:
```suggestion
                capture_exception(e, additional_properties={"tag": "sync_feature_flag_last_called", "flags_count": len(flags_to_update)})
```

haacked committed:
Provide context for captured exceptions

haacked committed:
Add test of naive timestamps from clickhouse

haacked committed:
Fix timezone comparison for ClickHouse timestamps

ClickHouse returns naive datetime objects but Django DateTimeField uses
timezone-aware datetimes. Comparing them raises TypeError.

Ensure timestamps from ClickHouse are converted to timezone-aware before
comparison with Django model fields.
